drivers: sensor: npm1300: Added charging support without NTC

jasr93 opened the pull request:
Added  charging support without NTC for npm1300 charger driver

MaureenHelm commented:
@jasr93 please fix the compliance check.

@anangl please take a look

nordic-auko commented on drivers/sensor/nordic/npm1300_charger/npm1300_charger.c:
```diff
@@ -456,6 +456,7 @@ int npm1300_charger_init(const struct device *dev)
 {
 	const struct npm1300_charger_config *const config = dev->config;
 	uint16_t idx;
+	uint8_t byte = 0U;
```
Nitpick: more descriptive name for this variable.

nordic-auko reviewed (approved):
Looks good. Testing and it is working.

Not required, but nice to have: return -ENOTSUP for `SENSOR_CHAN_GAUGE_TEMP` in `npm1300_charger_channel_get()` when NTC is disabled.

jaz1-nordic reviewed (commented):
Also merge commits into one.

rajeev1986 commented:
Any plans to get this merge into main? I was thinking this issue is already fixed by Nordic in NCS 2.7.0 but found out that it's not and is still pending. Only realized this after seeing the charger not working on our device with nPM1300.
 Thank you all.

jasr93 commented:
Compliance check has been fixed.
Added code suggestion too.

Commit:
drivers: sensor: npm1300: Added charging support without NTC

Current nPM1300 charger driver does not work with batteries
without NTC thermistor. Added supported for this feature.

Signed-off-by: Javier Santos <jasr93@outlook.es>
